modules/nf-scil/preproc/n4/main.nf

```diff
@@ -22,22 +22,30 @@ process PREPROC_N4 {
     def bspline_knot_per_voxel = task.ext.bspline_knot_per_voxel ? "$task.ext.bspline_knot_per_voxel" : "1"
     def shrink_factor = task.ext.shrink_factor ? "$task.ext.shrink_factor" : "1"
     """
-    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=$task.cpus
+    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
     export OMP_NUM_THREADS=1
     export OPENBLAS_NUM_THREADS=1
     export ANTS_RANDOM_SEED=1234
 
-    spacing=\$(mrinfo -spacing $b0 | tr " " "\\n" | sort -n | tail -1)
-    knot_spacing=\$(echo "\$spacing/$bspline_knot_per_voxel" | bc -l)
+    if [[ -f "$ref" ]]
+    then
+        spacing=\$(mrinfo -spacing $ref | tr " " "\\n" | sort -n | tail -1)
+        knot_spacing=\$(echo "\$spacing/$bspline_knot_per_voxel" | bc -l)
 
-    N4BiasFieldCorrection -i $b0\
-        -o [${prefix}__b0_n4.nii.gz, bias_field_b0.nii.gz]\
-        -c [300x150x75x50, 1e-6] -v 1\
-        -b [\${knot_spacing}, 3] \
-        -s $shrink_factor
+        N4BiasFieldCorrection -i $ref\
+            -o [${prefix}__ref_n4.nii.gz, bias_field_ref.nii.gz]\
```

Comment:
This gets included in the `image` output of the module. Either change the name here or specify the glob for the output further.

I am not against outputting this, but it should be done in another optional output channel.